content/client-lib-development-guide/features.textile

```diff
@@ -482,11 +482,11 @@ h3(#realtime-presence). Presence
 
 h3(#eventemitter). EventEmitter mixin / interface
 
-* @(RTE1)@ @EventEmitter@ is a generic interface for event registration and delivery used in a number of the types in the Realtime client library.  For example, the @Connection@ object emits events for connection state using the @EventEmitter@ pattern
+* @(RTE1)@ @EventEmitter@ is a generic interface for event registration and delivery used in a number of the types in the Realtime client library. For example, the @Connection@ object emits events for connection state using the @EventEmitter@ pattern
 * @(RTE2)@ Where objects provide @subscribe@ or @unsubscribe@ methods, they should follow the specification for the @EventEmitter#on@ and @EventEmitter#off@ methods respectively
-* @(RTE3)@ @EventEmitter#on@ registers a listener for the specified event if provided, else it is registered for all events
-* @(RTE4)@ @EventEmitter#once@ registers a listener that is called only once when the first of the applicable events occurs (ie the specified event if given, or the first event if it is registered for all events)
-* @(RTE5)@ @EventEmitter#off@ deregisters a listener for the specified event if given, else it is deregisters that listener for all registrations (global and event-specific)
+* @(RTE3)@ @EventEmitter#on@ allows a listener to be registered for either all events when no event argument is provided, or for only a single type of event when an event argument is provided. If @on@ is called twice with the exact same arguments, two listeners are added to the array of registered listeners. Therefore, it is perfectly valid to register the same listener multiple times and expect the listener to be called multiple times for each event emitted
+* @(RTE4)@ @EventEmitter#once@ registers a listener that is called only once when the first of the applicable events occurs (i.e. the specified event if given, or the first event if it is registered for all events). As with @on@, it is valid to register the same listener more than once which will result in the listener being called once for each registration
+* @(RTE5)@ @EventEmitter#off@ deregisters every matching listener for the specified event if given, else it is deregisters every matching listener for any registrations i.e. registrations for any event and registrations for a specific event. If @off@ is called with no arguments, all listeners for all and any events are removed
```

Comment:
Following up the above comment, what about "if called with a specific event and a listener, it removes all registrations that match both the given listener and the given event; if called only with a listener, it removes all registrations matching the given listener, no matter their associated specific event or if there is one; if called with no arguments, it removes all registrations, for all events and listeners."